packages/dashboard/src/components/queryEditor/iotSiteWiseQueryEditor/assetModelDataStreamExplorer/assetModelDataStreamExplorer.tsx

```diff
@@ -111,17 +102,8 @@ export const AssetModelDataStreamExplorer = ({
    */
   useEffect(() => {
     updateSelectedAsset(selectedAsset[0]);
-  }, [updateSelectedAsset, selectedAsset]);
-
-  useEffect(() => {
-    if (currentSelectedAsset)
-      selectAsset([
-        {
-          ...currentSelectedAsset,
-          assetId: currentSelectedAsset?.id,
-        } as AssetResource,
-      ]);
-  }, [currentSelectedAsset, selectAsset]);
+    selectAssetModelProperties([]); // if selectedAsset changes, dont keep state from old selection in RE
```

Comment:
should there be any checks around resetting the selected state?
is there any chance that something in the dependency array could change on accident?

Reply:
I dont think so. currentSelectedAsset and selectAssetModelProperties are constant functions, so the only thing that would result in re-render would be the selectedAsset itself. When a asset does change, we do want to clear whatever properties were selected from the previous asset, otherwise when the user clicks add it would add properties that don't appear selected on the screen.